MapzenSDK/MZMapViewController.swift

```diff
@@ -976,10 +986,31 @@ open class MZMapViewController: UIViewController, LocationManagerDelegate {
     setupTgControllerView()
     setupAttribution()
     setupFindMeButton()
+    observeLifecycleNotifications()
 
     tgViewController.gestureDelegate = self
     tgViewController.mapViewDelegate = self
     tgViewController.preferredFramesPerSecond = 60
+    tgViewController.pauseOnWillResignActive = true
+    tgViewController.resumeOnDidBecomeActive = true
+  }
+
+  deinit {
+    NotificationCenter.default.removeObserver(self)
+  }
+
+  //MARK: - Application Lifecycle Observance
+  func observeLifecycleNotifications(notificationCenter: NotificationCenterProtocol = NotificationCenter.default) {
+    notificationCenter.addObserver(self, selector: #selector(applicationWillResignActive), name: .UIApplicationWillResignActive, object: nil)
+    notificationCenter.addObserver(self, selector: #selector(applicationWillEnterForeground), name: .UIApplicationWillEnterForeground, object: nil)
```

Comment:
Why do we use `UIApplicationWillEnterForeground ` instead of `UIApplicationDidBecomeActive`?

Reply:
Hmmm I think I misread UIApplicationWillEnterForeground vs DidBecomActive description in the lifecycle docs. Basically I thought that DidBecomeActive would get called when the application became active regardless of foreground or background status. This can happen if your app uses background refresh via silent push notifications. In that scenario, we don't want the GPU to start rendering.

However, re-reading the docs (https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622956-applicationdidbecomeactive), it seems that DidBecomeActive is the appropriate time to restart GPU interaction. Will fix.